lightning/src/offers/invoice.rs

```diff
@@ -658,7 +659,11 @@ impl UnsignedBolt12Invoice {
 		let tlv_stream = TlvStream::new(&bytes).chain(TlvStream::new(&experimental_bytes));
 		let tagged_hash = TaggedHash::from_tlv_stream(SIGNATURE_TAG, tlv_stream);
 
-		Self { bytes, experimental_bytes, contents, tagged_hash }
+		let offer_id = match &contents {
+			InvoiceContents::ForOffer { .. } => Some(OfferId::from_valid_bolt12_tlv_stream(&bytes)),
+			InvoiceContents::ForRefund { .. } => None,
+		};
+		Self { bytes, experimental_bytes, contents, tagged_hash, offer_id }
```

Comment:
@jkczyz do you think that we should add in `OfferId` a `maybe_from_valid_bolt12_tlv_stream(&bytes) -> Option<OfferId>` to include the refund case without duplicate the `match` here?